File: tests/fixtures/python-simple/src/controllers/order_controller.py
"""Order controller — handles order-related HTTP endpoints."""

from ..payments.service import PaymentService
from ..payments.types import PaymentResult
from ..utils.logger import Logger


class OrderController:
    """Handles order creation and management."""

    def __init__(self, payment_service: PaymentService, logger: Logger):
        self._payment_service = payment_service
        self._logger = logger

    async def create_order(self, user_id: str, amount: float) -> PaymentResult:
        """Create a new order and process payment."""
        self._logger.info(f"Creating order for {user_id}")
        result = await self._payment_service.process_payment(amount, user_id)
        return result

    async def cancel_order(self, order_id: str) -> bool:
        """Cancel an existing order."""
        self._logger.info(f"Cancelling order {order_id}")
        return await self._payment_service.refund(order_id)

File: tests/fixtures/java-simple/src/main/java/com/example/controllers/OrderController.java
package com.example.controllers;

import com.example.payments.PaymentService;
import com.example.payments.PaymentResult;
import com.example.utils.Logger;

/**
 * Controller for order operations.
 */
public class OrderController {

    private final PaymentService paymentService;
    private final Logger logger;

    public OrderController(PaymentService paymentService, Logger logger) {
        this.paymentService = paymentService;
        this.logger = logger;
    }

    public PaymentResult createOrder(String orderId, double amount) {
        logger.info("Creating order: " + orderId);
        PaymentResult result = paymentService.processPayment(orderId, amount);
        return result;
    }

    public void cancelOrder(String orderId) {
        logger.info("Cancelling order: " + orderId);
    }
}
